app/utils/formInitialization.ts

```diff
@@ -0,0 +1,40 @@
+import type { FormValues, ValidationFunction } from '../types/signupForm';
+
+interface ConfigItem {
+  field: keyof FormValues;
+  validate?: ValidationFunction;
+}
+
+/**
+ * Generates initial form values based on the provided config.
+ * @param config - Array of ConfigItem objects.
+ * @returns Object with field names as keys and empty strings as values.
+ */
+export const generateInitialValues = (config: ConfigItem[]): FormValues => {
+  const initialValues = {} as FormValues;
```

Comment:
For type safety, I would use annotation rather than assertion here: 
`const initialValues: FormValues = {};`